build.gradle.kts

```diff
@@ -386,3 +386,13 @@ tasks.register<Zip>("allIncubationReportsZip") {
 }
 
 fun Project.collectAllIncubationReports() = subprojects.flatMap { it.tasks.withType(IncubatingApiReportTask::class) }
+
+// Ensure the archives produced are reproducible
+allprojects {
+    tasks.withType<AbstractArchiveTask>() {
+        isPreserveFileTimestamps = false
+        isReproducibleFileOrder = true
+        dirMode = 0775
+        fileMode = 0664
```

Comment:
I think we should use the same `fileMode`/`dirMode` as we use for the `wrapper.jar`: https://github.com/gradle/gradle/blob/ac75dd6f84d5c540d024aaf98df0ef49be82b5d2/subprojects/wrapper/wrapper.gradle#L42-L43

Reply:
Do you know why we use those modes for the wrapper JAR?

Reply:
Actually, that code should be removed now I guess.

Reply:
Yeah, the code should be removed. We used those modes since those were the modes we used when publishing from CI - i.e. our default modes on our Ubuntu machines.